lib/util/LocationStatusManager.js

```diff
@@ -563,15 +631,17 @@ class LocationStatusManager {
                 this._pushActionToRedis(service, location, actions.resume, schedule);
                 return next();
             }
-            const paused = this._locationStatusStore[location].getServicePauseStatus(service);
-            if (paused) {
-                this._locationStatusStore[location].resumeLocation(service, schedule);
+            const conditions = { paused: true };
+            const newStatus = { paused: !!schedule, scheduledResume: schedule || null };
```

Comment:
nit: maybe more "readable" with ternary?

```suggestion
            const newStatus = schedule ? { paused: true, scheduledResume: schedule }
                                       : { paused: false, scheduledResume: null };
```